app/models/user.rb

```diff
@@ -170,16 +170,11 @@ def notification_preferences
   end
 
   def accessible_services
-    services = account.accessible_services
+    account.accessible_services.permitted_for_user(self)
```

Comment:
This method already has a test 😄 
https://github.com/3scale/porta/blob/4f1fc50eac10b99825c28e4e7ca2b93103754dac/test/unit/user_test.rb#L88-L109